src/components/dialog/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC, PropsWithChildren } from 'react';
+
+import {
+  Dialog as ChakraDialog,
+  ConditionalValue,
+  Portal,
+  RecipeVariantProps,
+  UseDialogProps,
+  useSlotRecipe,
+} from '@chakra-ui/react';
+
+import { dialogRecipe } from 'src/themes/shared/slotRecipes/dialog';
+
+import { H3 } from '../heading';
+import { Button } from '../button';
+import { DialogCloseButton } from './DialogCloseButton';
+import { Box } from '../box';
+
+type MotionPreset = 'none' | 'scale';
+
+type DialogRootProps = RecipeVariantProps<typeof dialogRecipe>;
+
+type Sizes = 'md' | 'lg' | 'full' | 'auth0';
+
+type ActionButton = {
+  action: () => void;
+  label: string;
+};
+
+export interface DialogProps extends DialogRootProps {
+  title?: string;
+  open?: UseDialogProps['open'];
+  onOpenChange?: UseDialogProps['onOpenChange'];
+  motionPreset?: MotionPreset;
+  primaryActionButton?: ActionButton;
+  secondaryActionButton?: ActionButton;
+  variant?: 'fullScreen' | 'topScroll';
+  size?: Sizes;
+  disableBodyPadding?: boolean;
+  overlayColor?: 'gray';
+}
```

Comment:
Can we go for `type` for consistency:


```suggestion
export type DialogProps = DialogRootProps & {
  title?: string;
  open?: UseDialogProps['open'];
  onOpenChange?: UseDialogProps['onOpenChange'];
  motionPreset?: MotionPreset;
  primaryActionButton?: ActionButton;
  secondaryActionButton?: ActionButton;
  variant?: 'fullScreen' | 'topScroll';
  size?: Sizes;
  disableBodyPadding?: boolean;
  overlayColor?: 'gray';
}
```